src/rgw/rgw_data_sync.cc

```diff
@@ -4103,6 +4169,7 @@ class RGWBucketShardIncrementalSyncCR : public RGWCoroutine {
     set_status("init");
     rules = sync_pipe.get_rules();
     target_location_key = sync_pipe.info.dest_bs.bucket.get_key();
+    generation = sync_pipe.source_bucket_info.layout.logs.back().gen;
```

Comment:
@cbodley is this the right place to set a default value for generation?

Reply:
no, i don't think bucket sync should ever need to choose a default - it should always be getting an explicit generation number from the datalog. for this PR, it's fine to use gen=0 until the datalog part gets hooked up